src/middlewared/middlewared/api/base/server/ws_handler/rpc.py

```diff
@@ -49,7 +49,20 @@ def __init__(self, middleware: "Middleware", origin: ConnectionOrigin, ws: WebSo
         self.subscriptions = {}
 
     def send(self, data):
-        asyncio.run_coroutine_threadsafe(self.ws.send_str(json.dumps(data)), self.middleware.loop)
+        try:
+            data_ = json.dumps(data)
+        except Exception as e:
+            self.send_truenas_error(
+                data.get("id"),
+                JSONRPCError.INTERNAL_ERROR.value,
+                "Failed to JSON serialize server message",
+                None,
+                str(e),
+                sys.exc_info(),
+                str(data.get("result"))
```

Comment:
This might fail as well, leading to the same error. It's better not to send anything here. Also, the type is `list | None`, but we're passing string.